zone1/cli.c

```diff
@@ -0,0 +1,497 @@
+/* Copyright(C) 2018 Hex Five Security, Inc. - All Rights Reserved */
+
+#include <cli.h>
+#include <robot.h>
+#include <mzmsg.h>
+
+#define PRINT_BUFFER_SIZE   128
+static char print_buffer[PRINT_BUFFER_SIZE] = "";
+
+static const char welcome_msg[] =
+	"\e[2J\e[H" // clear terminal screen
+	"======================================================================\n"
+	"     	       Hex Five MultiZone(TM) Security v.0.1.1\n"
+	"    Copyright (C) 2018 Hex Five Security Inc. All Rights Reserved\n"
+	"======================================================================\n"
+	" This version of MultiZone(TM) is meant for evaluation purposes only.\n"
+	" As such, use of this software is governed by your Evaluation License.\n"
+	" There may be other functional limitations as described in the\n"
+	" evaluation kit documentation. The full version of the software does\n"
+	" not have these restrictions.\n"
+	"======================================================================\n"
+;
+
+static mzmsg_t zone2;
+
+void restart(){
+	ECALL_CSRC_MIE();
+	mzmsg_reset(&zone2);
+	asm ("j _start");
+}
+
+unsigned long handle_syncexception(unsigned long mcause, unsigned long mtval, unsigned long mepc){
+
+	uint32_t rst = 0;
+
+	switch(mcause){
+
+        case 0x0: // Instruction address misaligned
+            sprintf(print_buffer, "Instruction address misaligned : 0x%08x 0x%08x 0x%08x \n", mcause, mtval, mepc);
+			rst = 1;
+            break;
+        case 0x1: // Instruction access fault
+        	sprintf(print_buffer, "Instruction access fault : 0x%08x 0x%08x 0x%08x \n", mcause, mtval, mepc);
+			rst = 1;
+            break;
+        case 0x2: // Illegal Instruction
+            sprintf(print_buffer, "Illegal instruction : 0x%08x 0x%08x 0x%08x \n", mcause, mtval, mepc);
+			rst = 1;
+            break;
+        case 0x4: // Load address misaligned
+        	sprintf(print_buffer, "Load address misaligned : 0x%08x 0x%08x 0x%08x \n", mcause, mtval, mepc);   
+            break; 
+        case 0x5: // Load access fault
+            sprintf(print_buffer, "Load access fault : 0x%08x 0x%08x 0x%08x \n", mcause, mtval, mepc);
+            break;
+        case 0x6: // Store/AMO address misaligned
+            sprintf(print_buffer, "Store/AMO address misaligned : 0x%08x 0x%08x 0x%08x \n", mcause, mtval, mepc);
+            break;
+        case 0x7: // Store access fault
+            sprintf(print_buffer, "Store access fault : 0x%08x 0x%08x 0x%08x \n", mcause, mtval, mepc);
+            break;
+        default:
+			sprintf(print_buffer, "Unknown fault : 0x%08x 0x%08x 0x%08x \n", mcause, mtval, mepc);
+			rst = 1;
+            break;
+    }
+
+	mzmsg_write(&zone2, print_buffer, strlen(print_buffer));
+	
+	if(rst){
+		sprintf(print_buffer, "\nPress any key to restart");
+		mzmsg_write(&zone2, print_buffer, strlen(print_buffer));
+		char c='\0'; while(mzmsg_read(&zone2, &c, 1) == 0);
+		restart();
+	}
+
+	mepc += 4;
```

Comment:
Careful, this may not be always correct, there are some instructions that are 2 bytes!

Reply:
This would also explain why we get:
```
Illegal instruction : 0x00000002 0x0000fef4 0x204136ee
```
when attempting the `stats` command.